src/proxy/buffer.rs

```diff
@@ -1,28 +1,36 @@
-extern crate tower_buffer;
+extern crate linkerd2_router as rt;
 
-use std::{error, fmt, marker::PhantomData};
+use std::{fmt, marker::PhantomData};
 
-pub use self::tower_buffer::Buffer;
+use futures::{Future, Poll};
+use tower::buffer::{Buffer};
 
 use logging;
 use svc;
 
-/// Wraps `Service` stacks with a `Buffer`.
+type Error = Box<dyn std::error::Error + Send + Sync>;
+
+/// Produces `MakeService`s where the output `Service` is wrapped with a `Buffer`
 #[derive(Debug)]
 pub struct Layer<Req> {
     capacity: usize,
     _marker: PhantomData<fn(Req)>,
 }
 
-/// Produces `Service`s wrapped with a `Buffer`
+/// Produces `MakeService`s where the output `Service` is wrapped with a `Buffer`
 #[derive(Debug)]
-pub struct Stack<M, Req> {
+pub struct MakeBuffer<M, Req> {
```

Comment:
This is fairly minor but I notice there are some cases where these are named after what they make (i.e. `MakeBuffer`, `MakePending`, other cases where they're named `Make` or `MakeSvc`, and still others where they're still named `Stack`. Would be nice if this was more consistent.

Reply:
I swayed back and forth on what they should be called, and eventually stopped caring that some are still called `Stack`. 🤷‍♂️

Reply:
yeah, that's fair. i don't really care what they end up being called (though I'm somewhat in favour of the more information-rich `MakeWhatever`) but they should probably eventually be consistent?